openhands/runtime/base.py

```diff
@@ -213,6 +213,37 @@ async def on_event(self, event: Event) -> None:
             source = event.source if event.source else EventSource.AGENT
             self.event_stream.add_event(observation, source)  # type: ignore[arg-type]
 
+    def clone_repo(self, github_token: str | None, selected_repository: str | None):
+        if not github_token or not selected_repository:
+            return
+        url = f'https://{github_token}@github.com/{selected_repository}.git'
+        dir_name = selected_repository.split('/')[1]
+        action = CmdRunAction(
+            command=f'git clone {url} {dir_name} ; cd {dir_name} ; git checkout -b openhands-workspace'
+        )
+        logger.info(action, extra={'msg_type': 'ACTION'})
+        self.run_action(action)
+
+    def get_custom_microagents(self, selected_repository: str | None) -> list[str]:
+        if not selected_repository:
+            return []
+        dir_name = Path(selected_repository.split('/')[1])
+        files = self.list_files(str(dir_name))
+        logger.info(files)
+        custom_microagents_dir = Path('.openhands') / 'microagents'
```

Comment:
for backward compatibility, can we look for a file called `.openhands_instructions`, and make that a microagent that gets added to the system message (or the first user message)? It won't have any of the YAML at the top of the file--just markdown about the repository